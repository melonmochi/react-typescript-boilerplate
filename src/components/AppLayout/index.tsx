import * as React from "react";
import { Layout, Menu, Breadcrumb, Icon } from "antd";
import "./style";

const { SubMenu } = Menu;
const { Header, Content, Sider } = Layout;

const AppLayout = () => (
  <Layout>
    <Header>
      <Menu
        className="layout-header-menu"
        theme="dark"
        mode="horizontal"
        defaultSelectedKeys={["1"]}
      >
        <Menu.Item key="1">nav 1</Menu.Item>
        <Menu.Item key="2">nav 2</Menu.Item>
        <Menu.Item key="3">nav 3</Menu.Item>
      </Menu>
    </Header>
    <Layout>
      <Sider width={200} style={{ background: "#fff" }}>
        <Menu
          className="layout-sider-menu"
          mode="inline"
          defaultSelectedKeys={["1"]}
          defaultOpenKeys={["sub1"]}
        >
          <SubMenu
            key="sub1"
            title={
              <span>
                <Icon type="user" />
                subnav 1
              </span>
            }
          >
            <Menu.Item key="1">option1</Menu.Item>
            <Menu.Item key="2">option2</Menu.Item>
            <Menu.Item key="3">option3</Menu.Item>
            <Menu.Item key="4">option4</Menu.Item>
          </SubMenu>
        </Menu>
      </Sider>
      <Layout className="layout-center">
        <Breadcrumb className="layout-center-breadcrumb">
          <Breadcrumb.Item>Home</Breadcrumb.Item>
          <Breadcrumb.Item>List</Breadcrumb.Item>
          <Breadcrumb.Item>App</Breadcrumb.Item>
        </Breadcrumb>
        <Content className="layout-center-content">Content</Content>
      </Layout>
    </Layout>
  </Layout>
);

export default AppLayout;
