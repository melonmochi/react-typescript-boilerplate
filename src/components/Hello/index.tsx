import * as React from "react";
import { Button } from "antd";

export interface HelloProps {
  compiler: string;
  framework: string;
}

const Hello = (props: HelloProps) => (
  <h1>
    <Button type="primary">Button</Button>
    Hello from {props.compiler} and {props.framework}!
  </h1>
);

export default Hello;
