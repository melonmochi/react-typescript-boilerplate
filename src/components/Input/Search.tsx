import React, { Component } from "react";
import classNames from "classnames";
import { AutoComplete, Icon, Input } from "antd";
import { AutoCompleteProps, DataSourceItemType } from "antd/es/auto-complete";
import styles from "./Search.less";

export interface SearchProps {
  onPressEnter: (value: string) => void;
  onSearch: (value: string) => void;
  onChange: (value: string) => void;
  onVisibleChange: (b: boolean) => void;
  className: string;
  placeholder: string;
  defaultActiveFirstOption: boolean;
  dataSource?: DataSourceItemType[];
  defaultOpen: boolean;
  open?: boolean;
  defaultValue?: string;
}

interface SearchState {
  value?: string;
  searchMode: boolean;
}

class Search extends Component<SearchProps, SearchState> {
  static defaultProps = {
    defaultActiveFirstOption: false,
    onPressEnter: () => {},
    onSearch: () => {},
    onChange: () => {},
    className: "",
    placeholder: "",
    defaultOpen: false,
    onVisibleChange: () => {}
  };
  private inputRef: Input | null = null;

  constructor(props: SearchProps) {
    super(props);
    this.state = {
      searchMode: props.defaultOpen,
      value: props.defaultValue
    };
  }

  enterSearchMode = () => {
    const { onVisibleChange } = this.props;
    onVisibleChange(true);
    this.setState({ searchMode: true }, () => {
      const { searchMode } = this.state;
      if (searchMode && this.inputRef) {
        this.inputRef.focus();
      }
    });
  };

  getRefValue = (node: Input) => {
    this.inputRef = node;
  };

  leaveSearchMode = () => {
    this.setState({
      searchMode: false
    });
  };

  onKeyDown = (e: React.KeyboardEvent) => {
    if (e.key === "Enter") {
      console.log("im entering");
    }
  };

  onChange: AutoCompleteProps["onChange"] = value => {
    if (typeof value === "string") {
      const { onSearch, onChange } = this.props;
      this.setState({ value });
      onSearch(value);
      onChange(value);
    }
  };

  render() {
    const { className, defaultValue, placeholder } = this.props;
    const { searchMode, value } = this.state;
    const inputClass = classNames(styles.input, { [styles.show]: searchMode });
    return (
      <span
        className={classNames(className, styles.search)}
        onClick={this.enterSearchMode}
      >
        <Icon type="search" key="Icon" />
        <AutoComplete
          className={inputClass}
          key="Header Search AutoComplete"
          onChange={this.onChange}
          value={value}
        >
          <Input
            defaultValue={defaultValue}
            aria-label={placeholder}
            onBlur={this.leaveSearchMode}
            onKeyDown={this.onKeyDown}
            placeholder={placeholder}
            ref={this.getRefValue}
          />
        </AutoComplete>
      </span>
    );
  }
}

export default Search;
