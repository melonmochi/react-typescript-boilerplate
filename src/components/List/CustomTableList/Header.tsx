import React, { FC, Dispatch } from "react";
import { Form, Switch, Radio } from "antd";
import { RadioChangeEvent } from "antd/lib/radio";
import { TableListState, showHeader, expandedRowRender } from ".";
import { useTranslation } from "react-i18next";

const scroll = { y: 240 };

interface HeaderProps {
  state: TableListState;
  setState: Dispatch<React.SetStateAction<TableListState>>;
}

export const Header: FC<HeaderProps> = props => {
  const { state, setState } = props;
  const { t } = useTranslation();

  const title = () => t("mock.title");
  const footer = () => t("mock.footer");

  const handleToggle = (prop: any) => (enable: boolean) => {
    setState({ ...state, [prop]: enable });
  };

  const handleSizeChange = (e: RadioChangeEvent) => {
    setState({ ...state, size: e.target.value });
  };

  const handleTableLayoutChange = (e: RadioChangeEvent) => {
    setState({ ...state, tableLayout: e.target.value });
  };

  const handleExpandChange = (enable: boolean) => {
    setState({
      ...state,
      expandedRowRender: enable ? expandedRowRender : undefined
    });
  };

  const handleEllipsisChange = (enable: boolean) => {
    setState({ ...state, ellipsis: enable });
  };

  const handleTitleChange = (enable: boolean) => {
    setState({ ...state, title: enable ? title : undefined });
  };

  const handleHeaderChange = (enable: boolean) => {
    setState({ ...state, showHeader: enable ? showHeader : false });
  };

  const handleFooterChange = (enable: boolean) => {
    setState({ ...state, footer: enable ? footer : undefined });
  };

  const handleRowSelectionChange = (enable: boolean) => {
    setState({ ...state, rowSelection: enable ? {} : undefined });
  };

  const handleScollChange = (enable: boolean) => {
    setState({ ...state, scroll: enable ? scroll : undefined });
  };

  const handleDataChange = (hasData: boolean) => {
    setState({ ...state, hasData });
  };

  const handlePaginationChange = (e: RadioChangeEvent) => {
    const { value } = e.target;
    setState({
      ...state,
      pagination: value === "none" ? false : { position: value }
    });
  };

  return (
    <Form
      layout="inline"
      className="components-table-demo-control-bar"
      style={{ marginBottom: 16 }}
    >
      <Form.Item label={t("tableConfig.bordered")}>
        <Switch checked={state.bordered} onChange={handleToggle("bordered")} />
      </Form.Item>
      <Form.Item label={t("tableConfig.loading")}>
        <Switch checked={state.loading} onChange={handleToggle("loading")} />
      </Form.Item>
      <Form.Item label={t("tableConfig.title")}>
        <Switch checked={!!state.title} onChange={handleTitleChange} />
      </Form.Item>
      <Form.Item label={t("tableConfig.columnHeader")}>
        <Switch checked={!!state.showHeader} onChange={handleHeaderChange} />
      </Form.Item>
      <Form.Item label={t("tableConfig.footer")}>
        <Switch checked={!!state.footer} onChange={handleFooterChange} />
      </Form.Item>
      <Form.Item label={t("tableConfig.expandable")}>
        <Switch
          checked={!!state.expandedRowRender}
          onChange={handleExpandChange}
        />
      </Form.Item>
      <Form.Item label={t("tableConfig.checkbox")}>
        <Switch
          checked={!!state.rowSelection}
          onChange={handleRowSelectionChange}
        />
      </Form.Item>
      <Form.Item label={t("tableConfig.fixedHeader")}>
        <Switch checked={!!state.scroll} onChange={handleScollChange} />
      </Form.Item>
      <Form.Item label={t("tableConfig.hasData")}>
        <Switch checked={!!state.hasData} onChange={handleDataChange} />
      </Form.Item>
      <Form.Item label={t("tableConfig.ellipsis")}>
        <Switch checked={!!state.ellipsis} onChange={handleEllipsisChange} />
      </Form.Item>
      <Form.Item label={t("tableConfig.size.title")}>
        <Radio.Group value={state.size} onChange={handleSizeChange}>
          <Radio.Button value="default">
            {t("tableConfig.size.default")}
          </Radio.Button>
          <Radio.Button value="middle">
            {t("tableConfig.size.middle")}
          </Radio.Button>
          <Radio.Button value="small">
            {t("tableConfig.size.small")}
          </Radio.Button>
        </Radio.Group>
      </Form.Item>
      <Form.Item label={t("tableConfig.tableLayout.title")}>
        <Radio.Group
          value={state.tableLayout}
          onChange={handleTableLayoutChange}
        >
          <Radio.Button value={undefined}>
            {t("tableConfig.tableLayout.unset")}
          </Radio.Button>
          <Radio.Button value="fixed">
            {t("tableConfig.tableLayout.fixed")}
          </Radio.Button>
        </Radio.Group>
      </Form.Item>
      <Form.Item label={t("tableConfig.pagination.title")}>
        <Radio.Group
          value={state.pagination ? state.pagination.position : "none"}
          onChange={handlePaginationChange}
        >
          <Radio.Button value="top">
            {t("tableConfig.pagination.top")}
          </Radio.Button>
          <Radio.Button value="bottom">
            {t("tableConfig.pagination.bottom")}
          </Radio.Button>
          <Radio.Button value="both">
            {t("tableConfig.pagination.both")}
          </Radio.Button>
          <Radio.Button value="none">
            {t("tableConfig.pagination.none")}
          </Radio.Button>
        </Radio.Group>
      </Form.Item>
    </Form>
  );
};
