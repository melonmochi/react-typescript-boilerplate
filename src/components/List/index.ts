export { default as TableList } from "./TableList";
export { BigDataTableList } from "./BigDataTableList";
