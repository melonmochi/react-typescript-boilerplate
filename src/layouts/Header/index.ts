export { default as Dropdown } from "./Dropdown";
export { default as LanguageDropdown } from "./LanguageDropdown";
export { default as RightContent } from "./RightContent";
