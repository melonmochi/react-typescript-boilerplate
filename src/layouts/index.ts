export { default as BasicLayout } from "./BasicLayout";
export { default as Footer } from "./Footer";
