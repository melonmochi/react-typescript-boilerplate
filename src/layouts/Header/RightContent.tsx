import React from "react";
import { Icon, Tooltip } from "antd";
import { Search } from "@/components/Input";
import { LanguageDropdown } from ".";
import { ConnectProps } from "typings";
import styles from "./RightContent.less";

export type SiderTheme = "light" | "dark";
export type SiderLayout = "sidemenu" | "topmenu";
export interface GlobalHeaderRightProps extends ConnectProps {
  theme?: SiderTheme;
  layout?: SiderLayout;
}

const RightContent: React.SFC<GlobalHeaderRightProps> = props => {
  const { theme, layout } = props;
  let className = styles.right;

  if (theme === "dark" && layout === "topmenu") {
    className = `${styles.right}  ${styles.dark}`;
  }

  return (
    <div className={className}>
      <Search
        className={`${styles.action} ${styles.search}`}
        defaultActiveFirstOption={false}
        defaultOpen={false}
        // tslint:disable-next-line:jsx-no-lambda
        onChange={value => {
          console.log("change", value);
        }}
        // tslint:disable-next-line:jsx-no-lambda
        onSearch={value => {
          console.log("input", value);
        }}
        // tslint:disable-next-line:jsx-no-lambda
        onPressEnter={value => {
          console.log("enter", value);
        }}
        placeholder="Search"
      />
      <Tooltip title="Help">
        <a
          target="_blank"
          href="https://github.com/melonmochi/react-typescript-boilerplate"
          className={styles.action}
        >
          <Icon type="question-circle-o" />
        </a>
      </Tooltip>
      <LanguageDropdown className={styles.action} />
    </div>
  );
};

export default RightContent;
