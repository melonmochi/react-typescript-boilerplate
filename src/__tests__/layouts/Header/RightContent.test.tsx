import * as React from "react";
import { shallow } from "enzyme";
import { RightContent } from "@/layouts/Header";
import { Search } from "@/components/Input";
import { SiderTheme, SiderLayout } from "@/layouts/Header/RightContent";

describe("RightContent", () => {
  describe("Default props", () => {
    const container = shallow(<RightContent />);
    test("Snapshot", () => {
      expect(container).toMatchSnapshot();
    });

    describe("Search", () => {
      const search = container.find(Search);
      test("onChange", () => {
        search.props().onChange("text");
      });
      test("onSearch", () => {
        search.props().onSearch("text");
      });
      test("onPressEnter", () => {
        search.props().onPressEnter("text");
      });
    });
  });

  describe("Alternative props", () => {
    const props = {
      theme: "dark" as SiderTheme,
      layout: "topmenu" as SiderLayout
    };
    const container = shallow(<RightContent {...props} />);
    test("Snapshot", () => {
      expect(container).toMatchSnapshot();
    });
  });
});
