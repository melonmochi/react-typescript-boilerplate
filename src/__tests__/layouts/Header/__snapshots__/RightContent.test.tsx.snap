// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RightContent Alternative props Snapshot 1`] = `
<div
  className="undefined  undefined"
>
  <Search
    className="undefined undefined"
    defaultActiveFirstOption={false}
    defaultOpen={false}
    onChange={[Function]}
    onPressEnter={[Function]}
    onSearch={[Function]}
    onVisibleChange={[Function]}
    placeholder="Search"
  />
  <Tooltip
    arrowPointAtCenter={false}
    autoAdjustOverflow={true}
    mouseEnterDelay={0.1}
    mouseLeaveDelay={0.1}
    placement="top"
    title="Help"
    transitionName="zoom-big-fast"
  >
    <a
      href="https://github.com/melonmochi/react-typescript-boilerplate"
      target="_blank"
    >
      <Icon
        type="question-circle-o"
      />
    </a>
  </Tooltip>
  <LanguageDropdown />
</div>
`;

exports[`RightContent Default props Snapshot 1`] = `
<div>
  <Search
    className="undefined undefined"
    defaultActiveFirstOption={false}
    defaultOpen={false}
    onChange={[Function]}
    onPressEnter={[Function]}
    onSearch={[Function]}
    onVisibleChange={[Function]}
    placeholder="Search"
  />
  <Tooltip
    arrowPointAtCenter={false}
    autoAdjustOverflow={true}
    mouseEnterDelay={0.1}
    mouseLeaveDelay={0.1}
    placement="top"
    title="Help"
    transitionName="zoom-big-fast"
  >
    <a
      href="https://github.com/melonmochi/react-typescript-boilerplate"
      target="_blank"
    >
      <Icon
        type="question-circle-o"
      />
    </a>
  </Tooltip>
  <LanguageDropdown />
</div>
`;
